tenable/io/scans.py

```diff
@@ -563,10 +565,42 @@ def configure(self, scan_id, **kw):
         current = self.details(scan_id)
         updated = self._create_scan_document(kw)
         scan = dict_merge(current, updated)
-
+        scan = self.upsert_aws_credentials(scan)
         # Performing the actual call to the API with the updated scan record.
         return self._api.put('scans/{}'.format(scan_id),
-                    json=scan).json()
+                             json=scan).json()
+
+    def upsert_aws_credentials(self, scan):
+        '''
+        Checks the credential dict of scan dict to derive operation add or edit.
+        This function assumes there are no edit credentials in the credentials dict.
+        If there is any edit credentials,it would override the same.
+        Args:
+            scan: scan object to update edit credential if it matches criteria
+        Returns:
+            :obj:`dict`:
+                The scan with updated credentials.
+        '''
+        if 'credentials' in scan:
+            aws_existing_credential_id = None
+            if 'current' in scan['credentials'] \
+                    and 'Cloud Services' in scan['credentials']['current'] \
+                    and 'Amazon AWS' in scan['credentials']['current']['Cloud Services']:
+                aws_existing_credential = scan['credentials']['current']['Cloud Services']['Amazon AWS']
+                if len(aws_existing_credential) == 1:
+                    aws_existing_credential_id = aws_existing_credential[0]['id']
+            if 'add' in scan['credentials'] \
+                    and 'Cloud Services' in scan['credentials']['add'] \
+                    and 'Amazon AWS' in scan['credentials']['add']['Cloud Services']:
+                aws_new_credential = scan['credentials']['add']['Cloud Services']['Amazon AWS']
+                aws_new_credential_id = None
+                if len(aws_new_credential) == 1:
+                    aws_new_credential_id = aws_new_credential[0]['id']
```

Comment:
Reading id from the 1st element of the list by assuming id would present in AWS credentials always as per recommendation from @mvalliammal , if Id doesn't present in AWS credentials, code would break.

Reply:
Yes, as per tenable, amazon AWS credentials will always has a field called ID